capstone/src/main/webapp/script.js

```diff
@@ -176,46 +187,110 @@ function callback(results, status) {
   _showBlackOwnedBusiness = false;
 }
 
+/** Creates an animated marker for each result location
+   @param {Object} place location to set marker for
+ */
 function setMarker(place) {
   const marker = new google.maps.Marker({
-    map: map,
+    map: _map,
     position: place.geometry.location,
     animation: google.maps.Animation.DROP,
   });
-  addToDisplayPanel(place.name);
+  addToDisplayPanel(place);
 }
- 
-function addToDisplayPanel(name) {
-  const locationElement = document.getElementById("restaurant-results");
-  locationElement.appendChild(createLocationElement(name));
+
+/** Itemizes each result into the collapsible panel
+@param {Object} place location to add to results panel
+ */
+function addToDisplayPanel(place) {
+  // put call to get details here and get new place before adding to panel
+  const locationElement = document.getElementById('restaurant-results');
+  locationElement.appendChild(createLocationElement(place));
+  locationElement.appendChild(createAdditionalInfo(place));
 }
- 
-function createLocationElement(placeName) {
-  const locElement = document.createElement('p');
-  locElement.className = 'place';
- 
-  const mainElement = document.createElement('span');
-  mainElement.innerText = placeName;
- 
-  locElement.appendChild(mainElement);
-  return locElement;
+
+/** Toggles on and off additional info upon click */
+function displayAdditionalInfo() {
+  const coll = document.getElementsByClassName('collapsible');
+
+  for (let i = 0; i < coll.length; i++) {
+    coll[i].addEventListener('click', function() {
+      this.classList.toggle('active');
+      const content = this.nextElementSibling;
+      if (content.style.maxHeight) {
+        content.style.maxHeight = null;
+      } else {
+        content.style.maxHeight = content.scrollHeight + 'px';
+      }
+    });
+  }
 }
 
+/** Creates button with location name for each place
+    @param {Object} place location to create button for
+
+    @return {HTMLButtonElement} the created location button
+ */
+function createLocationElement(place) {
+  const mainElement = document.createElement('button');
+  mainElement.className = 'collapsible';
+  mainElement.innerHTML = place.name;
+
+  displayAdditionalInfo();
```

Comment:
Incidentally, every time we call `addEventListener()` it adds it adds a new event listener to the list of event listeners run on `click`.

This is probably the source of the bug you were running into earlier. Since here and on line 264 we're adding new event listeners to _every_ button, every time we add _a_ button, we're probably adding multiple click handlers. If we end up with an odd number of click handlers it'll end open after they all run, and if we end up with an even number of click handlers then it'll end up closed after they all run.